### PR TITLE
Ensure DropPrimaryKeyTask respects dry-run

### PR DESCRIPTION
**Pull request by lukedegruchy**
Closes https://github.com/hapifhir/hapi-fhir/issues/6061

### CODE REVIEW

**Review by TipzCM — APPROVED** [review]

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropPrimaryKeyTask.java [hunk 1]
```diff
@@ -51,35 +54,51 @@ public DropPrimaryKeyTask(String theProductVersion, String theSchemaVersion, Str
...
-		};
-
 		@Nullable
 		@Language("SQL")
 		final String primaryKeyNameSql = generatePrimaryKeyNameSql();
 
 		@Nullable
 		final String primaryKeyName = primaryKeyNameSql != null
-				? executeSqlWithResult(primaryKeyNameSql, resultSetExtractor, getTableNameWithDatabaseExpectedCase())
+				? executeSqlWithResult(primaryKeyNameSql, getTableNameWithDatabaseExpectedCase())
```
Comment (michaelabuckley) [inline hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropPrimaryKeyTask.java:63]: problem: This is confused.  We DO want to run this query during a dry run - i.e. fetching the table name.  It is only the actual DROP command we don't want to run.  That's why we shouldn't be using methods named `execute*` for querying the metatdata.

Please just replace this with something simple like
```suggestion
				? newJdbcTemplate().queryForObject("some sql", String.clasS); 
```

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropPrimaryKeyTask.java [hunk 2]
```diff
@@ -51,35 +54,51 @@ public DropPrimaryKeyTask(String theProductVersion, String theSchemaVersion, Str
...
 				: null;
 
 		ourLog.debug("primaryKeyName: {} for driver: {}", primaryKeyName, getDriverType());
 
 		return generateDropPrimaryKeySql(primaryKeyName);
 	}
 
+	private String executeSqlWithResult(@Language("SQL") String theSql, Object... theArguments) {
```
Comment (michaelabuckley) [inline hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropPrimaryKeyTask.java:71]: problem: delete this.  Dry-run is about the modifications, not building the sql.

**Review by michaelabuckley — CHANGES_REQUESTED** [review]
This is too complicated.  There are two parts to a migration like this:

1. query the metadata to build the migration sql
2. execute the migration sql

We only want dry-run to apply to 2, not to 1.

**Review by michaelabuckley — APPROVED** [review]
Much better.

**Pull request merged**